src/js_native_api_v8.cc

```diff
@@ -197,13 +197,15 @@ class RefBase : protected Finalizer, RefTracker {
        : Finalizer(env, finalize_callback, finalize_data, finalize_hint),
         _refcount(initial_refcount),
         _delete_self(delete_self) {
-    Link(finalize_callback == nullptr
-        ? &env->reflist
-        : &env->finalizing_reflist);
+    if (finalize_callback == nullptr) {
+      env->reflist.PushFront(this);
+    } else {
+      env->finalizing_reflist.PushFront(this);
+    }
```

Comment:
The mystery deepens – if I comment this out, the performance degradation disappears – even though this only runs once per process, and outside the `bench.start()`/`bench.end()` bracket 🤨

Reply:
@gabrielschulhof That’s very odd indeed… I could only imagine it being either be an issue with code generation (e.g. inlining) or a memory caching issue, but neither of them really make sense here.

Also, with GCC 9.2.1 on x64 Linux I can’t reproduce the performance impact myself.

Reply:
@addaleax I am using Ubuntu 19.10 with gcc 9.2.1-9ubuntu2 on Intel(R) Xeon(R) Platinum 8280 CPU @ 2.70GHz. I was thinking that Some Things™ get optimized out when that code is commented, and not otherwise 🤷